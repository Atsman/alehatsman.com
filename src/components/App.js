import App, { Container } from 'next/app'
import React from 'react'
import { GlobalStyle } from 'styles/global'
import Router from 'next/router'
import withGA from 'next-ga'

class MyApp extends App {
  static async getInitialProps ({ Component, ctx }) {
    return {
      pageProps: {
        ...(Component.getInitialProps ? await Component.getInitialProps(ctx) : {})
      }
    }
  }

  render () {
    const { Component, pageProps } = this.props
    return (
      <Container>
        <React.Fragment>
          <GlobalStyle />
          <Component {...pageProps} />
        </React.Fragment>
      </Container>
    )
  }
}

export default withGA('UA-99002751-2', Router)(MyApp)
