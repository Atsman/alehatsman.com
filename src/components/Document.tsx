import { ServerStyleSheet } from 'styled-components'
import Document, { Head, Main, NextScript, DocumentProps } from 'next/document'
import React from 'react'

interface Props extends DocumentProps {
  styleTags: any
}

export default class MyDocument extends Document<Props> {
  static async getInitialProps (ctx) {
    const sheet = new ServerStyleSheet()
    const originalRenderPage = ctx.renderPage

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: (App) => (props) => sheet.collectStyles(<App {...props} />)
        })

      const initialProps = await Document.getInitialProps(ctx)
      return {
        ...initialProps,
        styles: <>{initialProps.styles}{sheet.getStyleElement()}</>
      }
    } finally {
      sheet.seal()
    }
  }

  render () {
    return (
      <html>
        <Head>
          <link rel='stylesheet' type='text/css' href='/static/css/typography.css' />
          <link rel='icon' href='/static/images/logo.png' />
          <title>Aleh Atsman - Personal Blog</title>
          {this.props.styleTags}
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </html>
    )
  }
}
