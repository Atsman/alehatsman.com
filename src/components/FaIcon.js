import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import styled from 'styled-components'

const FIcon = (props) => (
  <FontAwesomeIcon {...props} />
)

export default styled(FIcon)``
