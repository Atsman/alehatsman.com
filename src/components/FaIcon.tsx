import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import styled from '@emotion/styled'

const FIcon = props => <FontAwesomeIcon {...props} fixedWidth />

export default styled(FIcon)``
