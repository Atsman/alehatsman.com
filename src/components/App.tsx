import App from 'next/app'
import React from 'react'
import Router from 'next/router'
import withGA from 'next-ga'

import { GlobalStyle } from '@/styles/global'

class MyApp extends App {
  static async getInitialProps ({ Component, ctx }) {
    return {
      pageProps: {
        ...(Component.getInitialProps
          ? await Component.getInitialProps(ctx)
          : {})
      }
    }
  }

  render () {
    const { Component, pageProps } = this.props
    return (
      <>
        <GlobalStyle />
        <Component {...pageProps} />
      </>
    )
  }
}

let myApp
if (process.env.NODE_ENV !== 'development') {
  myApp = withGA(process.env.GA_ID, Router)(MyApp)
} else {
  myApp = MyApp
}

export default myApp
