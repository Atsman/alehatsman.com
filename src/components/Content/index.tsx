import React from 'react'
import { MDXProvider } from '@mdx-js/react'

import components from './Content.styled'

interface Props {
  children: React.ReactNode
}

const CodeBlock = ({ children }: Props) => (
  <MDXProvider components={components}>{children}</MDXProvider>
)

export default CodeBlock
