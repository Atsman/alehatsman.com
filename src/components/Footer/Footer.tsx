import React, { FC } from 'react'

import { Flex } from '@/components/Box'

export const Footer: FC = ({ children }) => (
  <Flex justifyContent="center" borderTop={1} borderColor="gray.0" py={3}>
    {children}
  </Flex>
)
