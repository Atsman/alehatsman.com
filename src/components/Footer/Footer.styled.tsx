import styled from '@emotion/styled'
import FaIcon from '@/components/FaIcon'
import colors from '@/styles/colors'

export const FooterWrapper = styled.footer`
  display: flex;
  justify-content: center;

  padding: 20px;
  border-top: 1px solid ${colors.dimGray};
`

export const Nav = styled.nav`
  display: flex;
  flex-direction: row;
`

export const NavItem = styled.a`
  padding: 5px;
  ${FaIcon} {
    margin-right: 10px;
  }
  font-size: 20px;
`

export const NavSep = styled.span`
  display: flex;
  align-items: center;
  margin: 0 20px;
`
