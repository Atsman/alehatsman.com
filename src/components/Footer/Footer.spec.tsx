import React from 'react'
import { render } from '@testing-library/react'
import { Footer } from '@/components/Footer'

describe('Footer', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Footer />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly', () => {
    const { getByText } = render(<Footer>Content</Footer>)
    expect(getByText('Content')).toBeInTheDocument()
  })
})
