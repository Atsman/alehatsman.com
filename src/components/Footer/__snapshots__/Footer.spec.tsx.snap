// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  border-top: 1px;
  border-color: gray.0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-top: 16px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="emotion-0 emotion-1"
  />
</DocumentFragment>
`;
